pkg/tfgen/docs.go

```diff
@@ -503,6 +503,8 @@ func parseExamples(language language, lines []string) ([]string, bool, error) {
 	for i := 0; i < len(lines); i++ {
 		line := lines[i]
 		if strings.Index(line, "```") == 0 {
+			result = append(result, "{{% examples %}}")
```

Comment:
@cnunciato just as an FYI (and in case you have any comments), this is the short-code that the examples in the resource docs would be wrapped-in.

Reply:
Excellent, thanks for the mention. Having a look now.

Reply:
Ok yeah, I think this'll work fine. Having the `{{% examples %}}` shortcode will allow us to render the internal content as markdown, and then from there, we can add interactivity with components. https://github.com/pulumi/docs/pull/2665 shows an example.

Reply:
One nice-to-have would be if you could pass the language to the shortcode as well (so like `{{% examples typescript %}}`), if that made sense. That'd make it a bit easier to wrap them all in a language chooser without having to dig into the rendered markup to figure that out.

Reply:
We'll also want to capture the example titles, when we have them -- probably just as markdown just above the backticked code samples, e.g.:

```
    {{% examples %}}

    ### Some title

    ```typescript
    import foo from "bar";
    ```

    ### Some other title

    ```typescript
    import bar from "baz";
    ```

    ### Some third title

    ```typescript
    import qux from "quux";
    ```

    {{% /examples %}}
```

Reply:
For my understanding, what is the reason for passing a language to the short-code?

Also, the short-code will actually wrap _every_ single example, where each example will contain snippets in the supported languages. So your example above is actually like this:
```
    {{% examples %}}
    ### Some title for the example

    ```typescript
    import foo from "bar";
    ...
    ```

    ```python
    from "bar" import foo
    ...
    ```

    ```go
    import github.com/....
    ...
    ```

    ```csharp
    using Foo.Bar;
    ...
    ```

    {{% /examples %}}
```

Reply:
Ah, okay -- I misread the code, then. It looked as though all of the code snippets within an `{{% examples %}}` block would be of the same language, but if that's not the case, passing the language at the top level wouldn't make sense, yeah. 

As I mentioned, the only reason would be to avoid having to discern the language of the whole lot by examining the HTML attributes of every snippet, which seemed unnecessary if they'd all be the same. But since they won't, just `{{% examples %}}` would be fine.

Reply:
Although one question @praneetloke, based on your snippet --

Will there be more than one heading within an `{{% examples%}}` shortcode? Or just one, with multiple `{{% examples %}}` shortcodes within a single document? In other words, would there be this:

```
{{% examples %}}

### Heading 

[some typescript]

[some go]

### Another heading

[some typescript]

[some go]

{{% /examples %}}
```

or this?

```
{{% examples %}}
### Heading 

[some typescript]

[some go]

{{% /examples %}}

{{% examples %}}
### Another heading 

[some typescript]

[some go]

{{% /examples %}}
```

Reply:
The reason I ask is, I think it's _most_ important to know where the _set_ of examples starts and ends, more than it is to have each individual example bookended, because I want to be able to wrap them all into a single component, so that each one can be expanded or collapsed individually. So wrapping them all in a single shortcode would be preferable, if that were possible.

Reply:
@cnunciato it would be "Snippet 2" from this [comment](https://github.com/pulumi/pulumi-terraform-bridge/pull/130#discussion_r393287427). I actually would also like confirmation from @tasiah as well on this.

Reply:
I think the goal is Snippet 2, like Praneet said. I'll have to go back and make sure that that's actually true though 😅 .

Reply:
Ah, then that's not ideal. How feasible is option 1?

Reply:
Shouldn't be too bad (🤞). I'll move the shortcodes around then unless @praneetloke or @pgavlin have any objections.

Reply:
We had an offline conversation about this. We _could_ do option 1, but we would also want to see if Pat has any reservations for that per his original suggestion to do it the way this PR has it currently (i.e. option 2). We will sync-up with Pat and update this PR accordingly.

Reply:
Something like this could also work (technically, at least):

```
{{% examples %}}

{{% example %}}

### Some example

[some typescript]

[some go]

{{% /example %}}

{{% example %}}

### Some OTHER example

[some typescript]

[some go]

{{% /example %}}

{{% /examples %}}
```

... but things do get a bit dicey with Hugo rendering in this case, as the tolerance for spacing between these nested shortcodes can be brittle and depending on positioning, produce inconsistent results. So from my perspective, it'd be ideal just to wrap all of the examples in one shortcode (as in my Snippet 1), and let that single shortcode and the client-side code figure out how to render them.

Reply:
> So from my perspective, it'd be ideal just to wrap all of the examples in one shortcode (as in my Snippet 1), and let that single shortcode and the client-side code figure out how to render them.

Ok we'll confirm with Pat and do this.